### DISCUSSION TITLE
VSCode Typescript formatter - Hard wrap at <n> characters

### DISCUSSION THREAD

**Discussion by alexandru-mng** [microsoft/vscode-discussions#1147] [Extension Development QnA] [answered]
Can the builtin typescript formatter split any line that is longer than a configurable number of characters?

For example, for a max line length of 80, the following line
```typescript
new Date().toLocaleDateString('en', { weekday: 'long', day: 'numeric', month: 'long', year: 'numeric' })
```
should be formatted as 
```typescript
new Date().toLocaleDateString('en',
      { weekday: 'long', day: 'numeric', month: 'long', year: 'numeric' })
```

**Comment by burkeholland** [comment]
Look like the built-in formatting is controlled by the `typescript.format.*` settings which have a pretty limited number of configuration options. The docs [also mention that it's pretty limited](https://code.visualstudio.com/Docs/languages/typescript#_formatting). Tagging @DanielRosenwasser here for help.

**Comment by DanielRosenwasser** [comment, marked as answer]
Hi @alexandru-mng, right now a maximum line length is something that the built-in TypeScript formatter doesn't support. The closest issue on our side tracking this is https://github.com/microsoft/TypeScript/issues/22991, but we haven't prioritized it yet.

In the meantime, there are alternative formatters you can try out. [Prettier](https://prettier.io/) is a popular formatter that tends to be a bit more prescriptive, and [has a dedicated VS Code extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode). [dprint](https://dprint.dev/) is a another formatter that our team uses happily. It's fast and has a lot of flexibility in configuration, and also has [a VS Code extension](https://marketplace.visualstudio.com/items?itemName=dprint.dprint).
